src/clr-angular/utils/focus/focus-on-view-init/focus-on-view-init.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { AfterViewInit, Directive, ElementRef, HostListener, Inject, Input, PLATFORM_ID } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+
+/* This directive is for guiding the document focus to the new added content when its view is initialized. */
+@Directive({
+  selector: '[clrFocusOnViewInit]',
+  host: {
+    '[attr.tabindex]': 'tabIndex',
+  },
+})
+export class ClrFocusOnViewInit implements AfterViewInit {
+  constructor(private el: ElementRef, @Inject(PLATFORM_ID) private platformId: Object) {}
+
+  // negative tabindex as we shouldn't have this in the tab order
+  private _tabIndex: number = -1;
+
+  // set to true as we will intend to set the tab index by default
+  // if the element the directive is placed on has tabindex already, we will use that existing one.
+  private tabIndexSet: boolean = true;
+
+  get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  @Input('tabindex')
+  set tabIndex(value: number) {
+    const numValue = Number(value);
+    if (typeof numValue === 'number') {
+      this._tabIndex = numValue;
+      this.tabIndexSet = false;
+    } else {
+      this._tabIndex = -1;
+    }
+  }
+
+  @HostListener('focusout')
+  onFocusout() {
+    // Once the focusout is registered, we should remove the tabindex we set.
+    // As it has already served its purpose during view initialization.
```

Comment:
I'm not sure if I understand this correctly. Should the element have tabindex='-1' after initialization and all of the lifecycles have run or does this remove it when the focusout event is executed? 

When I run the dev app I'm not seeing tabindex on the focused element after navigating back and forth (http://localhost:4200/vertical-nav/accessibility/beatles) but the test for adding -1 passes. I do see it when I refresh the page though so I'm not sure I understand how the directive should behave.

Reply:
The goal of the directive is to set the focus on its host element once the directive's view gets initialized. And once the user moves the focus to somewhere by clicking with her mouse or pressing the tab key, tabindex='-1' should be removed because our goal is not to make the element focusable, but to facilitate screen reader to catch new rendered content/title/header. 

As for not finding the tabindex, I think you are clicking with your mouse to inspect elements in the web inspection tool, the focus will be lost from the directive element so the `focusout` is being fired on the directive resulting in the removal of the tabindex.

Reply:
I see, yeah I can't seem to replicate the loss of tabindex so it must have been removing it when I inspected. 

The final visual difference I noticed here is that every element this is placed on seems to get a focus ring indicating it is focused, if this is only for screen readers should we handle mitigating the visual change or is it intended to have the focus ring.